src/provider/provider_tracking.c

```diff
@@ -119,6 +120,134 @@ static umf_result_t trackingAlloc(void *hProvider, size_t size,
     return ret;
 }
 
+static umf_result_t trackingAllocationSplit(void *provider, void *ptr,
```

Comment:
Given that we use a tree for tracking, it might be simpler to avoid preemptively splitting/merging pages, and instead handle that during inserts/removals.

Reply:
it might be error-prone concerning IPC handle support. I prefer to do it explicitly when a pool manager calls merge/split functions (like implemented right now).
In general, it is an interesting case, the IPC handle is created per coarse-grain allocation (returned by the memory provider) and if two coarse-grain allocations are merged what should we do with corresponding IPC handles if them were created before? I will think more about that. I know that L0 just does not support split/merge functionality but tracking memory provider should work for all cases.

Reply:
Yeah, I was thinking about this as well. Doing this lazily in removal would definitely be simpler (and probably faster) but as Sergey said, might be problematic for IPC. I think we can revisit this once we have IPC support merged.